x-pack/filebeat/input/http_endpoint/handler.go

```diff
@@ -92,22 +96,47 @@ func httpReadJsonObject(body io.Reader) (obj common.MapStr, status int, err erro
 		return nil, http.StatusInternalServerError, fmt.Errorf("failed reading body: %w", err)
 	}
 
-	if !isObject(contents) {
-		return nil, http.StatusBadRequest, errUnsupportedType
-	}
-
-	obj = common.MapStr{}
-	if err := json.Unmarshal(contents, &obj); err != nil {
+	var jsBody interface{}
+	if err := json.Unmarshal(contents, &jsBody); err != nil {
 		return nil, http.StatusBadRequest, fmt.Errorf("malformed JSON body: %w", err)
 	}
 
-	return obj, 0, nil
+	switch v := jsBody.(type) {
+	case map[string]interface{}:
+		objs = append(objs, v)
+	case []interface{}:
+		for idx, obj := range v {
+			asMap, ok := obj.(map[string]interface{})
+			if !ok {
+				return nil, http.StatusBadRequest, fmt.Errorf("%v at index %d", errUnsupportedType, idx)
+			}
+			objs = append(objs, asMap)
+		}
+	default:
+		return nil, http.StatusBadRequest, errUnsupportedType
+	}
+	return objs, 0, nil
 }
 
-func isObject(b []byte) bool {
-	obj := bytes.TrimLeft(b, " \t\r\n")
-	if len(obj) > 0 && obj[0] == '{' {
-		return true
+func httpReadNDJSON(body io.Reader) (objs []common.MapStr, status int, err error) {
```

Comment:
I might be misreading this, but could we just have one function?  I think with json.NewDecoder we could handle:

```
{"a": 1}
or
{"a": 1}{"b":2}
or
{"a":1}
{"b":2}
or
[{"a":1},{"b":2}]
```
and then json or ndjson would both work.

Reply:
Something like this:
``` go
func httpReadJSON(body io.Reader) (objs []common.MapStr, status int, err error) {
	if body == http.NoBody {
		return nil, http.StatusNotAcceptable, errBodyEmpty
	}

	decoder := json.NewDecoder(body)
	for idx := 0; ; idx++ {
		var obj interface{}
		if err := decoder.Decode(&obj); err != nil {
			if err == io.EOF {
				break
			}
			return nil, http.StatusBadRequest, errors.Wrapf(err, "malformed JSON object at stream position %d", idx)
		}
		switch v := obj.(type) {
		case map[string]interface{}:
			objs = append(objs, v)
		case []interface{}:
			for listIdx, listObj := range v {
				asMap, ok := listObj.(map[string]interface{})
				if !ok {
					return nil, http.StatusBadRequest, fmt.Errorf("%v at obj index %d, list index %d", errUnsupportedType, idx, listIdx)
				}
				objs = append(objs, asMap)
			}
		default:
			return nil, http.StatusBadRequest, errUnsupportedType
		}
	}
	return objs, 0, nil
}
```

I can push it to the PR if you want.